xbmc/Application.cpp

```diff
@@ -2778,6 +2779,26 @@ bool CApplication::OnMessage(CGUIMessage& message)
     }
     break;
 
+    case GUI_MSG_PLAY_TRAILER:
+    {
+      const CFileItemPtr item = std::static_pointer_cast<CFileItem>(message.GetItem());
+      CFileItem* trailerItem =
+          ContentUtils::GeneratePlayableTrailerItem(item, g_localizeStrings.Get(20410));
+
+      if (item->IsPlayList())
+      {
+        CFileItemList* l = new CFileItemList; //don't delete,
+        l->Add(std::make_shared<CFileItem>(*trailerItem));
+        CServiceBroker::GetAppMessenger()->PostMsg(TMSG_MEDIA_PLAY, -1, -1, static_cast<void*>(l));
+      }
+      else
+      {
+        CServiceBroker::GetAppMessenger()->PostMsg(TMSG_MEDIA_PLAY, 1, 0,
+                                                   static_cast<void*>(trailerItem));
```

Comment:
```suggestion
                                                   static_cast<void*>(trailerItem.get()));
```
should be thay to retrieve the raw ptr then

Reply:
Same as above, since when the `TMSG_MEDIA_PLAY` message is consumed the item is deleted, this will result in a crash when the shared/unique pointer goes out of scope and tries to free the raw pointer (already freed)

Reply:
Beside that `traileritem` is a raw pointer, it has no `get()`function.

Reply:
i meant if it _was_ a shared_ptr